pkg/endpointmanager/manager.go

```diff
@@ -414,31 +413,6 @@ func RegenerateAllEndpoints(owner regeneration.Owner, regenMetadata *regeneratio
 	return &wg
 }
 
-// RegenerateEndpointSet calls a SetStateLocked for each endpoint from the given
-// set 'epsToRegen' and if it still exists in the list of endpoints manager it
-// regenerates if state transaction is valid. During this process, the endpoint
-// list is locked and cannot be modified.
-// Returns a waiting group that can be used to know when all the endpoints are
-// regenerated.
-func RegenerateEndpointSet(owner regeneration.Owner, regenMetadata *regeneration.ExternalRegenerationMetadata, epsToRegen map[uint16]struct{}) *sync.WaitGroup {
-	mutex.RLock()
-	defer mutex.RUnlock()
-
-	var wg sync.WaitGroup
-	log.WithFields(logrus.Fields{"endpoints": epsToRegen}).Infof("regenerating some endpoints due to %s", regenMetadata.Reason)
-	for epID := range epsToRegen {
-		ep := endpoints[epID]
-		if ep != nil {
-			wg.Add(1)
-			go func(ep *endpoint.Endpoint) {
```

Comment:
This commit eliminates goroutines and yet doesn't mention this at all. Please split such functional changes out or at minimum, describe the change in the commit (& PR) descriptions, potential impact, maybe why they're not necessary?

Reply:
I'll write that since the `RegenerateEndpointSet` was replaced with `epsToRegen.ForEach` as it runs each endpoint in a different go routine the one initialized in `RegenerateEndpointSet` is no longer needed. Sounds good?

Reply:
Oh, I completely missed that the `ForEach()` makes its own goroutines :open_mouth: These are the kinds of subtleties that made me nervous through my whole review..

Thanks, yes, that would help a lot. (And maybe we can highlight that the `EndpointSet.ForEach()` is actually an `EndpointSet.ForEachGo()` / `EndpointSet.ForEachAsync()`...? I appreciate I'm diving out of the territory of changes in this PR, but this was another surprise to me.)